### PR TITLE
SOLR-16035: Enable spotless for prometheus-exporter

### PR DESCRIPTION
**Pull request by risdenk**
https://issues.apache.org/jira/browse/SOLR-16035

### CODE REVIEW

File: solr/prometheus-exporter/src/java/org/apache/solr/prometheus/exporter/SolrClientFactory.java [hunk 1]
```diff
@@ -57,13 +57,18 @@ public CloudSolrClient createCloudSolrClient(String zookeeperConnectionString) {
...
-            .collect(Collectors.toList()),
-        Optional.ofNullable(parser.getChrootPath()));
-
-    cloudBuilder.withConnectionTimeout(settings.getHttpConnectionTimeout())
+    CloudSolrClient.Builder cloudBuilder =
+        new CloudSolrClient.Builder(
+            parser.getServerAddresses().stream()
+                .map(
+                    address ->
+                        String.format(
```
Comment (dsmiley) [inline solr/prometheus-exporter/src/java/org/apache/solr/prometheus/exporter/SolrClientFactory.java:65]: Simple string concatenation with a colon would be simpler and shorter, probably wouldn't wrap

File: solr/prometheus-exporter/src/java/org/apache/solr/prometheus/exporter/SolrExporter.java [hunk 2]
```diff
@@ -144,39 +157,68 @@ private SolrScraper createScraper(SolrScrapeConfiguration configuration, Prometh
...
       case CLOUD:
         return new SolrCloudScraper(
-            factory.createCloudSolrClient(configuration.getZookeeperConnectionString().get()), requestExecutor, factory);
+            factory.createCloudSolrClient(configuration.getZookeeperConnectionString().get()),
+            requestExecutor,
+            factory);
       default:
-        throw new RuntimeException(String.format(Locale.ROOT, "Invalid type: %s", configuration.getType()));
+        throw new RuntimeException(
+            String.format(Locale.ROOT, "Invalid type: %s", configuration.getType()));
```
Comment (dsmiley) [inline solr/prometheus-exporter/src/java/org/apache/solr/prometheus/exporter/SolrExporter.java:165]: String concatenation would be shorter; probably wouldn't wrap

File: solr/prometheus-exporter/src/test/org/apache/solr/prometheus/collector/MetricSamplesTest.java [hunk 3]
```diff
@@ -25,46 +28,48 @@
...
-
-    assertEquals(expectedValues, test1.samples.stream().map(s -> s.value).collect(Collectors.toList()));
+    Collector.MetricFamilySamples test1 =
+        allMetrics.stream()
+            .filter(s -> s.name.equals(metricName))
+            .findFirst()
+            .orElseThrow(
+                () ->
+                    new RuntimeException(
+                        String.format(Locale.ROOT, "Unable to find item %s", metricName)));
```
Comment (dsmiley) [inline solr/prometheus-exporter/src/test/org/apache/solr/prometheus/collector/MetricSamplesTest.java:58]: Simple string concatenation probably wouldn't have wrapped

Comment (risdenk) [reply solr/prometheus-exporter/src/test/org/apache/solr/prometheus/collector/MetricSamplesTest.java]: Fixed in 6f59d6fa54dd1711174bd1e607b7c14e54fa6cd7

File: solr/prometheus-exporter/src/java/org/apache/solr/prometheus/exporter/SolrExporter.java (hunk 2 above)
Comment (risdenk) [reply solr/prometheus-exporter/src/java/org/apache/solr/prometheus/exporter/SolrExporter.java]: Fixed in 6f59d6fa54dd1711174bd1e607b7c14e54fa6cd7

File: solr/prometheus-exporter/src/java/org/apache/solr/prometheus/exporter/SolrClientFactory.java (hunk 1 above)
Comment (risdenk) [reply solr/prometheus-exporter/src/java/org/apache/solr/prometheus/exporter/SolrClientFactory.java]: Fixed in 6f59d6fa54dd1711174bd1e607b7c14e54fa6cd7

**Review by dsmiley — APPROVED** [review]

**Pull request merged**